contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/TwoCountriesCriterion.java

```diff
@@ -50,7 +55,7 @@ private boolean filterBranch(Terminal terminal1, Terminal terminal2) {
         }
         Country countrySide1 = substation1.getCountry().orElse(null);
         Country countrySide2 = substation2.getCountry().orElse(null);
-        if (countrySide1 == null || countrySide2 == null) {
+        if (countrySide1 == null && !countries1.isEmpty() || countrySide2 == null && !countries2.isEmpty()) {
```

Comment:
Same thing: make sure in the constructor that the lists are not null

Reply:
Done